controller_manager/src/controller_manager.cpp

```diff
@@ -838,6 +777,15 @@ controller_interface::return_type ControllerManager::switch_controller(
   const std::vector<std::string> & deactivate_controllers, int strictness, bool activate_asap,
   const rclcpp::Duration & timeout)
 {
+  if (!resource_manager_->is_urdf_already_loaded())
+  {
+    RCLCPP_ERROR(
+      get_logger(),
+      "Resource Manager is not initialized yet! Please provide robot description on "
+      "'~/robot_description' topic before trying to switch controllers.");
+    return controller_interface::return_type::ERROR;
+  }
+
```

Comment:
```suggestion
```

This might not be needed as the controller manager services will only exist once the resource manager is initialized, so this would be redundant

Reply:
you are right. I'll remove it

Reply:
Actually, there is a test testing that. I would keep it, if someone extends controller manager directly. It is just a little protection to not do stupid things.